### PR TITLE
Add public API for reading CAF config files

### PR DESCRIPTION
**Pull request by Neverlord**

### CODE REVIEW

**Comment by josephnoir** [comment]
There are no consequences to parsing a config, right? It simply gives you the settings to inspect (provided the config is well formed).

**Comment by Neverlord** [comment]
> There are no consequences to parsing a config, right?

You mean side effects? No. This simply gives users the possibility to use CAF-style config files without having to go through an `actor_system_config`.

File: libcaf_core/caf/actor_system_config.hpp [hunk 1]
```diff
@@ -331,6 +331,63 @@ class actor_system_config {
...
+                                 settings& result);
+
+  /// Parses the content of `source` using CAF's config format.
+  /// @param source Character sequence in CAF's config format.
+  /// @returns A ::settings dictionary with the parsed content of `source` on
+  ///          success, an ::error otherwise.
+  static expected<settings> parse_config(std::istream& source);
+
+  /// Parses the content of `source` using CAF's config format. Also
+  //  type-checks user-defined parameters in `opts`.
```
Comment (josephnoir) [inline libcaf_core/caf/actor_system_config.hpp]: Missing `/`.

**Comment by Neverlord** [comment]
> What happens if x

@josephnoir is there more to this sentence?

**Comment by josephnoir** [comment]
No, sorry. I was about to ask the question above in the review comment and forgot to delete the text before.

**Pull request merged**